src/node_crypto.cc

```diff
@@ -4637,12 +4653,23 @@ void ECDH::SetPublicKey(const FunctionCallbackInfo<Value>& args) {
   EC_POINT* pub = ecdh->BufferToPoint(Buffer::Data(args[0].As<Object>()),
                                       Buffer::Length(args[0].As<Object>()));
   if (pub == nullptr)
-    return;
+    return env->ThrowError("Failed to convert Buffer to EC_POINT");
 
   int r = EC_KEY_set_public_key(ecdh->key_, pub);
   EC_POINT_free(pub);
   if (!r)
-    return env->ThrowError("Failed to convert BN to a private key");
+    return env->ThrowError("Failed to set EC_POINT as the public key");
+}
+
+bool ECDH::IsKeyPairValid() {
+  return 1 == EC_KEY_check_key(key_);
+}
+
+void ECDH::ValidKeyPair(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  ECDH* ecdh = Unwrap<ECDH>(args.Holder());
+  Local<Boolean> result = Boolean::New(env->isolate(), ecdh->IsKeyPairValid());
+  args.GetReturnValue().Set(result);
```

Comment:
There is an overload for `bool`, you can just `args.GetReturnValue().Set(ecdh->IsKeyPairValid())` here.